0000-bury-description.md

```diff
@@ -0,0 +1,49 @@
+- Feature Name: optional_error_description
+- Start Date: 2017-11-29
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Default implementation of `Error::description()`
+[summary]: #summary
+
+Provide a default implementation of the `Error` trait's `description()` method to save users trouble of implementing this flawed method.
+
+# Motivation
+[motivation]: #motivation
+
+The `description()` method is a waste of time for implementors and users of the `Error` trait. There's high overlap between description and `Display`, which creates redundant implementation work and confusion about relationship of these two ways of displaying the error.
+
+The `description()` method can't easily return a formatted string with per-instance error description. That's a gotcha for novice users struggling with the borrow checker, and gotcha for users trying to display the error, because the `description()` is going to return a less informative message than the `Display` trait.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Let's steer users away from the `description()` method.
+
+1. Change the `description()` documentation to suggest use of the `Display` trait instead.
+2. Provide a default implementation of the `description()` so that the `Error` trait can be implemented without worrying about this method.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The description method can have default implementation returning `core::intrinsics::type_name::<Self>()`. This preserves basic functionality of this method for backwards compatibility.
```

Comment:
Are we comfortable somewhat stabilizing `type_name`? This makes it at least somewhat easy to (abuse) this to get it's functionality on stable for non-error types in at least binaries where the impl of Error doesn't matter too much from an exposed API perspective.

Either way, we should explicitly state that this method's default return value *is not stable* and should not be relied on from version to version.

Reply:
Simon's current PR just returns `""`